### PR TITLE
Added planetary-computer

### PR DESCRIPTION
**Pull request by TomAugspurger**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

The current release on PyPI did not include a license file. That's being fixed in https://github.com/microsoft/planetary-computer-sdk-for-python/pull/6. In the meantime, I've included the license file here.

### CODE REVIEW

File: planetary-computer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+about:
+  home: https://pypi.org/project/planetary-computer/
+  summary: Planetary Computer SDK for Python
+  license: MIT
+  license_file: LICENSE
+
+extra:
+  recipe-maintainers:
+    - TomAugspurger
+    - lossyrob
```
Comment (TomAugspurger) [inline planetary-computer/meta.yaml:47]: @lossyrob when you get a chance can you +1 to indicate that you're OK with being a maintainer here?

**Comment by TomAugspurger** [comment]
Looks like planetary-computer has some work to do to support building from source. I'll get that fixed and then update here.

File: planetary-computer/meta.yaml (hunk 1 above)
Comment (lossyrob) [reply planetary-computer/meta.yaml]: @TomAugspurger good by me!

**Comment by ocefpaf** [comment]
> Looks like planetary-computer has some work to do to support building from source. I'll get that fixed and then update here.

I was about to send a PR upstream but it seems you are ready to issue a new release. Let me know when you need a new review here.

**Comment by ocefpaf** [comment]
@TomAugspurger looks like the PR that fixes the sdist is in upstream's source. Do you want to mint a new release or should we patch things here? I prefer the former but the latter is fine if a new release is going to take a while.

**Comment by lossyrob** [comment]
@ocefpaf Tom's on a well deserved vaca - would a new pypi release of `planetary-computer` do the trick here? I can tag a new release if so

**Comment by TomAugspurger** [comment]
OK it took a bit but this should be ready @ocefpaf!

**Comment by ocefpaf** [comment]
Sorry. I was the one "out of office" last week. Thanks!

Pinging @rsignell-usgs b/c this was the last missing piece in your env :stuck_out_tongue_winking_eye:

**Pull request merged**